src/posit/connect/external/aws.py

```diff
@@ -13,12 +13,34 @@
 
 def get_aws_credentials(client: Client, user_session_token: str) -> Dict[str, str]:
     """
-    Get AWS credentials using OAuth token exchange.
+    Get AWS credentials using OAuth token exchange for an AWS Viewer integration.
 
     According to RFC 8693, the access token must be a base64 encoded JSON object
     containing the AWS credentials. This function will decode and deserialize the
     access token and return the AWS credentials.
 
+    Examples
+    --------
+    ```python
+    from posit.connect import Client
+    from posit.connect.external.aws import get_aws_credentials
+    import boto3
+    from shiny.express import session
+
+    client = Client()
+    session_token = session.http_conn.headers.get("Posit-Connect-User-Session-Token")
+    credentials = get_aws_credentials(client, user_session_token)
+    aws_session_expiration = credentials["expiration"]
+    aws_session = boto3.Session(
+        aws_access_key_id=credentials["accessKeyId"],
+        aws_secret_access_key=credentials["secretAccessKey"],
+        aws_session_token=credentials["sessionToken"],
+    )
```

Comment:
We should return a `TypedDict` here and map the keys to the Session keyword arguments:

```python
class Credentials(TypedDict):
    aws_access_key_id: str
    aws_secret_access_key: str
    aws_session_token: str
```

This saves the users a few steps...

```suggestion
    credentials = get_aws_credentials(client, user_session_token)
    aws_session = boto3.Session(**credentials)
```

Reply:
probably can't just pass it through though since the credentials includes the expiration field which the sesison constructor does not know about and will error out

Reply:
Ah, that makes sense. Probably best not to tightly couple it anyway in case boto3 changes down the road. What are your thoughts on using the TypedDict with snake_case variable names and then keeping the rest the same?

Reply:
Yeah true good point. Yes will do that